include/deal.II/base/geometry_info.h

```diff
@@ -2924,6 +2924,15 @@ GeometryInfo<dim>::child_to_cell_coordinates (const Point<dim>         &/*p*/,
 
 
 
+template <int dim>
+inline
+bool
+GeometryInfo<dim>::is_inside_unit_cell (const Point<dim> &)
```

Comment:
There are specializations of this function for 1d, 2d, and 3d. So the general template will never be called -- and as a consequence, it can live in the .cc file just fine.

Reply:
We instantiate `GeometryInfo` explicitly for `dim=0,1,2,3,4` and the reported missing definitions are related to `dim=4`. 
Looking at the other functions defined it is probably more consistent to have a specialization for `dim=4` instead. Would you agree on that as well?

Reply:
OK. Having a third look, we really mix template specializations with general templates already. Where some of the general templates are just declaring `ExcNotImplemented` and others are actually used. In the end, I don't like to have some in the include files and others in the source file; this would just be confusing. Is it really all that bad to have everything in `geometry_info.h`?

Reply:
Not terrible in any specific case. But if you think of our compile times, then it clearly does make a difference.

If you feel adamant about putting these functions into the `.h` file, then I'm not going to object. But it seems like an unnecessary case of making our `.h` files even bigger.